.github/workflows/ci.yml

```diff
@@ -0,0 +1,103 @@
+name: Pipeline
+on: [push]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Start timer
+        id: start_timer
+        run: |
+         echo "START_TIME=$(date +%s)" >> $GITHUB_ENV
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '18'
```

Comment:
don't hardcode the version, developers typically modify package.json to include the version requirements
https://github.com/ErmakovDenis/angular-react-starter/blob/10061ddfa8cc63fd063a543c43bea7b97681ad0d/angular/package.json#L3